File: 0347-top-k-frequent-elements/solution.py
class Solution:
    from collections import heapq
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = Counter(nums)
        res = []
        heap = []
        for x in count:
            heapq.heappush(heap, (count[x] * -1, x))
        while len(res) < k:
            cur = heapq.heappop(heap)
            res.append(cur[1])
        return res

File: 0347-top-k-frequent-elements/solution.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        priority_queue<pair<int, int>> q;
        unordered_map<int, int> count;
        for(auto cur: nums){   
            count[cur] += 1;
        }
        pair<int, int> temp;
        for(auto x: count){
            temp = {x.second, x.first};
            q.push(temp);
        }
        int i = 0;
        vector<int> res;
        while(i < k){
            res.push_back(q.top().second);
            q.pop();
            i++;
        }
        return res;
    }
};
